src/main/java/gregtech/common/metatileentities/multi/MetaTileEntityFissionReactor.java

```diff
@@ -0,0 +1,1096 @@
+                                        fuelIn.setInternalFuelRod(component);
+                                    }
+                                    fuelIn.setInternalFuelRod(component);
+                                    fissionReactor.addComponent(component, i + radius - 1, j + radius - 1);
+                                    continue;
+                                }
+                            }
+                        }
+                        this.unlockAll();
+                        setLockingState(LockingState.MISSING_FUEL);
+                        return;
+                    } else if (mte instanceof IControlRodPort controlIn) {
+                        ControlRod component = new ControlRod(100000, controlIn.hasModeratorTip(), 1, 800);
+                        fissionReactor.addComponent(component, i + radius - 1, j + radius - 1);
+                    }
+                }
+            }
+        }
+        if (!foundFuel) {
+            this.unlockAll();
+            setLockingState(LockingState.NO_FUEL_CHANNELS);
+            return;
+        }
+        fissionReactor.prepareThermalProperties();
+        fissionReactor.computeGeometry();
+        setLockingState(LockingState.LOCKED);
+    }
+
+    private void updateReactorState(int flowRate) {
+        this.fissionReactor.updatePower();
+        this.fissionReactor.updateTemperature(flowRate);
+        this.fissionReactor.updatePressure();
+        this.fissionReactor.updateNeutronPoisoning();
+        this.fissionReactor.regulateControlRods();
+    }
+
+    protected void setLockingState(LockingState lockingState) {
+        if (this.lockingState != lockingState) {
+            writeCustomData(GregtechDataCodes.SYNC_LOCKING_STATE, (buf) -> buf.writeEnumValue(lockingState));
+        }
+        this.lockingState = lockingState;
+    }
+
+    @Override
+    public long getCoverCapacity() {
+        // power is in MW
+        return (int) (this.maxPower * 1e6);
+    }
+
+    @Override
+    public long getCoverStored() {
+        // power is in MW
+        return (int) (this.power * 1e6);
+    }
+
+    private enum LockingState {
+        // The reactor is locked
+        LOCKED,
+        // The reactor is unlocked
+        UNLOCKED,
+        // The reactor is supposed to be locked, but the locking logic is yet to run
+        SHOULD_LOCK,
+        // The reactor can't lock because it is missing fuel in a fuel channel
+        MISSING_FUEL,
+        // The reactor can't lock because it is missing coolant in a coolant channel
+        MISSING_COOLANT,
+        // The reactor can't lock because a fuel output is clogged
+        FUEL_CLOGGED,
+        // There are no fuel channels at all!
+        NO_FUEL_CHANNELS,
+        // The reactor can't lock because components are flagged as invalid
+        INVALID_COMPONENT
+    }
+
+    public List<MultiblockShapeInfo> getMatchingShapes() {
```

Comment:
Is this a decent reactor? If we are displaying a custom reactor for the JEI page, it should be somewhat decent.